### PR TITLE
Implement Two Factor Authentication for administrators

### PR DESCRIPTION
**Pull request by tgalopin**
Based on #93.

### CODE REVIEW

File: src/AppBundle/Admin/AdministratorAdmin.php [hunk 1]
```diff
@@ -61,6 +77,12 @@ function ($plain) use ($admin) {
                         }
                     ))
             );
+
+        if (!$isCreation) {
+            $formMapper->add('googleAuthenticatorSecret', null, [
```
Comment (HeahDude) [inline src/AppBundle/Admin/AdministratorAdmin.php]: Same comment as in the previous PR `null` as second argument is preferable.

File: src/AppBundle/Controller/Security/AdminSecurityController.php [hunk 2]
```diff
@@ -56,4 +57,15 @@ public function loginCheck()
...
     {
     }
+
+    /**
+     * QR-code generator to be used with Google Authenticator.
+     *
+     * @Route("/qr-code/{id}", name="app_admin_qrcode")
+     * @Method("GET")
+     */
+    public function qrcodeAction(Administrator $administrator): Response
```
Comment (HeahDude) [inline src/AppBundle/Controller/Security/AdminSecurityController.php]: `qrCodeAction`?

File: src/AppBundle/Controller/Security/AdminSecurityController.php [hunk 3]
```diff
@@ -56,4 +57,15 @@ public function loginCheck()
     public function logoutAction()
     {
     }
+
+    /**
+     * QR-code generator to be used with Google Authenticator.
+     *
+     * @Route("/qr-code/{id}", name="app_admin_qrcode")
```
Comment (HeahDude) [inline src/AppBundle/Controller/Security/AdminSecurityController.php]: `app_admin_qr_code`?

File: src/AppBundle/Entity/Administrator.php [hunk 4]
```diff
@@ -44,6 +45,13 @@ class Administrator implements UserInterface
     private $password;
 
     /**
+     * @var string|null
+     *
+     * @ORM\Column(nullable=true)
+     */
+    private $googleAuthenticatorSecret;
```
Comment (HeahDude) [inline src/AppBundle/Entity/Administrator.php]: Does it really need to be persisted, or just to be checked once to trust the user?

Maybe saving a date would be more useful, i.e to handle expiration before asking again.

Comment (tgalopin) [inline src/AppBundle/Entity/Administrator.php]: This is not the code, this is the secret used to generate the code. A bit like a private key. It's stored on the server and on the application only once.

Comment (HeahDude) [inline src/AppBundle/Entity/Administrator.php]: Understood, thanks.

**Pull request merged**